api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/DeleteIndividualController.java

```diff
@@ -0,0 +1,198 @@
+package edu.cornell.mannlib.vitro.webapp.controller.freemarker;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.io.StringWriter;
+import java.util.HashMap;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryExecutionFactory;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.QuerySolutionMap;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.shared.Lock;
+
+import edu.cornell.mannlib.vitro.webapp.auth.permissions.SimplePermission;
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.AuthorizationRequest;
+import edu.cornell.mannlib.vitro.webapp.controller.VitroRequest;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.RedirectResponseValues;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.ResponseValues;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.TemplateResponseValues;
+import edu.cornell.mannlib.vitro.webapp.dao.DisplayVocabulary;
+import edu.cornell.mannlib.vitro.webapp.dao.jena.event.BulkUpdateEvent;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelNames;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.ChangeSet;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFService;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.impl.RDFServiceUtils;
+
+@WebServlet(name = "DeleteIndividualController", urlPatterns = "/deleteIndividualController")
+public class DeleteIndividualController extends FreemarkerHttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	private static final Log log = LogFactory.getLog(DeleteIndividualController.class);
+	private static final boolean BEGIN = true;
+	private static final boolean END = !BEGIN;
+
+	private static String TYPE_QUERY = "" + "PREFIX vitro:    <http://vitro.mannlib.cornell.edu/ns/vitro/0.7#>"
+			+ "SELECT ?type " + "WHERE" + "{ ?individualURI vitro:mostSpecificType ?type ." + "}";
+	private static String queryForDeleteQuery = "PREFIX display: <" + DisplayVocabulary.DISPLAY_NS + "> \n"
+			+ "SELECT ?deleteQueryText WHERE { ?associatedURI display:hasDeleteQuery ?deleteQueryText }";
+
+	private static final String DEFAULT_DELETE_QUERY_TEXT = "DESCRIBE ?individualURI";
+
+	@Override
+	protected AuthorizationRequest requiredActions(VitroRequest vreq) {
+		return SimplePermission.DO_FRONT_END_EDITING.ACTION;
```

Comment:
This is something that a lot of existing controllers tend not to handle well, but which we should probably consider more closely.  It seems that this controller should require the user to have permission at least for edu.cornell.mannlib.vitro.webapp.auth.requestedAction.resource.DropResource for the main individualURI that is deleted.  The thornier question is what to do with the related resources and (possibly the specific statements connecting them).  Do we want to pre-run the deletion query and check each of the additional individual URIs it returns to see whether the user also has DropResource permissions on those?  If a user has permission to drop the main resource but not some of the related individuals, do we fail on the whole operation or allow them to delete those individuals they are able to while leaving the rest behind?